src/components/Link/IstioObjectLink.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { Paths } from '../../config';
+import { Link } from 'react-router-dom';
+import { Badge, Tooltip, TooltipPosition } from '@patternfly/react-core';
+import { IstioTypes } from '../VirtualList/Config';
+
+interface Props {
+  name: string;
+  namespace: string;
+  type: string;
+}
+
+const IstioObjectLink = (props: Props) => {
+  const { name, namespace, type } = props;
+  //const objectType = type.charAt(0).toUpperCase() + type.slice(1);
+  const istioType = IstioTypes[type];
+  const url = '/namespaces/' + namespace + '/' + Paths.ISTIO + '/' + istioType.slug + '/' + name;
```

Comment:
Note that this is not going to work for templates/adapters.
As kiali supports/supported the old legacy way, we have a
```
http://localhost:3000/console/namespaces/istio-system/istio/templates/instances/attributes
```
for these special cases: adapter/templates.

My suggestion, don't change it here, but just add a comment for the future.

Reply:
I changed the method itself to handle type and subtype. However the consumer component doesn't render type/subtype.